hooks/useSharedGuideIds.tsx

```diff
@@ -0,0 +1,58 @@
+import {
+  useState,
+  useEffect,
+  useContext,
+  createContext,
+  ReactNode,
+  Dispatch,
+  SetStateAction,
+} from "react";
+import { SHARED_GUIDES_LOCAL_STORAGE } from "utils/constants";
+
+const SHARED_GUIDES_INITIAL_STATE = [""];
+
+interface SharedGuideIdsContextResult {
+  sharedGuideIds: string[];
+  setSharedGuideIds: Dispatch<SetStateAction<string[]>>;
+}
+
+const SharedGuideIdsContext = createContext<SharedGuideIdsContextResult>(
+  {} as SharedGuideIdsContextResult
+);
+
+function getGuideIds() {
+  const rawSharedGuideIds = localStorage.getItem(SHARED_GUIDES_LOCAL_STORAGE);
+
+  const sharedGuideIds =
+    rawSharedGuideIds?.split("|") || SHARED_GUIDES_INITIAL_STATE;
+
+  return sharedGuideIds;
+}
+
+export function SharedGuidesProvider({ children }: { children: ReactNode }) {
```

Comment:
I feel like a context provider is unnecessary for this because local storage can provide the context to any component. A hook that accesses local storage should be enough

Reply:
Sure, that's valid. I'll simplify it